write-fonts/src/tables/variations.rs

```diff
@@ -21,6 +21,237 @@ impl VariationRegionList {
+        }
+    }
+}
+
+impl FontWrite for PackedDeltaRun<'_> {
+    fn write_into(&self, writer: &mut TableWriter) {
+        self.compute_flag().write_into(writer);
+        match self {
+            PackedDeltaRun::Zeros(_) => (),
+            PackedDeltaRun::OneByte(deltas) => {
+                deltas.iter().for_each(|v| (*v as i8).write_into(writer))
+            }
+            PackedDeltaRun::TwoBytes(deltas) => deltas.iter().for_each(|v| v.write_into(writer)),
+        }
+    }
+}
+
+impl crate::validate::Validate for PackedPointNumbers {
+    fn validate_impl(&self, ctx: &mut ValidationCtx) {
+        if self.numbers.len() > 0x7FFF {
+            ctx.report("length cannot be stored in 15 bites");
+        }
+    }
+}
+
+impl FontWrite for PackedPointNumbers {
+    fn write_into(&self, writer: &mut TableWriter) {
+        // compute the actual count:
+        if self.is_all {
+            0u8.write_into(writer);
+        } else if self.numbers.len() <= 127 {
+            (self.numbers.len() as u8).write_into(writer);
+        } else {
+            (self.numbers.len() as u16).write_into(writer);
+        }
+
+        for run in self.iter_runs() {
+            run.write_into(writer);
+        }
+    }
+}
+
+impl PackedPointNumbers {
+    /// Create new packed numbers from raw numbers.
+    ///
+    /// The `is_all` flag should be true if there is a number value for each
+    /// point in the corresponding glyph (or CVT value in the cvar table).
+    pub fn new(numbers: Vec<u16>, is_all: bool) -> Self {
+        Self { is_all, numbers }
+    }
+    fn iter_runs(&self) -> impl Iterator<Item = PackedPointRun> {
+        const U8_MAX: u16 = u8::MAX as u16;
+        const MAX_POINTS_PER_RUN: usize = 128;
+
+        let mut points = self.numbers.as_slice();
+        let mut prev_point = 0u16;
+
+        // split a run off the front of points:
+        // - if point is more than 255 away from prev, we're using words
+        std::iter::from_fn(move || {
+            let next = points.first()?;
+            let (run_len, are_words) = if (next - prev_point) > U8_MAX {
+                let count = points
+                    .iter()
+                    .take(MAX_POINTS_PER_RUN)
+                    .scan(prev_point, |prev, point| {
+                        let result = (point - *prev > U8_MAX).then_some(point);
+                        *prev = *point;
+                        result
+                    })
+                    .count();
+                (count, true)
+            } else {
+                let count = points
+                    .iter()
+                    .take(MAX_POINTS_PER_RUN)
+                    .scan(prev_point, |prev, point| {
+                        let result = (point - *prev <= U8_MAX).then_some(point);
+                        *prev = *point;
+                        result
+                    })
+                    .count();
+                (count, false)
+            };
```

Comment:
I don't think you need to duplicate the iteration if you compute are_words and use it to pick waht function to scan for?

```rust
let are_words = next - prev_point > U8_MAX;
let mut scan_fn = if are_words {
  |v| v > U8_MAX
} else {
  |v| v <= U8_MAX
}
let count = points
   .iter()
   .take(MAX_POINTS_PER_RUN)
   .scan(prev_point, |prev, point| {
   let result = scan_fn(point - *prev).then_some(point);
      *prev = *point;
      result
   })
   .count();
```